src/message_dao.py

```diff
@@ -150,3 +159,6 @@ def store_last_hash(self, last_hash: str) -> None:
 
     def get_last_hash(self) -> int:
         return self.last_hash
+
+    def previous_versions(self, message: Message) -> List[Message]:
```

Comment:
should this be called previous_version_candidates() as well?

Reply:
Whoops. Left over from the previous naming scheme. Changed accordingly.